### PR TITLE
Connect to API

### PR DESCRIPTION
**Pull request by FilbertKimasa**
In this pull request, the following was done:

- [ ] Make use of the Bookstore API as a backend
- [ ] Use Axios to make your HTTP requests
- [ ] Implement fetch book functionality
- [ ] Implement add book functionality
- [ ] implement delete book functionality

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -1,43 +1,81 @@
...
+    bookData,
+  );
+
+  if (response.status !== 200) {
+    throw new Error('Failed to add a book');
+  }
+
+  const addedBook = response.data;
+  return addedBook;
+});
```
Comment (Johnadibe) [inline src/redux/books/booksSlice.js]: > - [ ] I've observed that when attempting to add a book to the list, the update doesn't happen automatically, and a page refresh is required to see the changes. To enhance the user experience, could you please address this issue and ensure that the list updates seamlessly without the need for manual page refreshing? Your attention to this matter would be greatly appreciated. 👍
> 
> [screen-capture (1).webm](https://github.com/FilbertKimasa/book-store/assets/40009354/f3b16d5f-50e4-400e-ba7e-3fd6bfdaaf70)
>

File: src/redux/books/booksSlice.js [hunk 2]
```diff
@@ -1,43 +1,81 @@
...
+  const response = await axios.delete(
+    `https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/${appId}/books/${bookId.item_id}`,
+  );
+
+  if (response.status !== 200) {
+    throw new Error('Failed to remove the book');
+  }
+
+  return bookId;
+});
```
Comment (Johnadibe) [inline src/redux/books/booksSlice.js]: > - [ ] I also observed that when attempting to delete a book from the list, the update doesn't happen automatically, and a page refresh is required to see the changes. To enhance the user experience, could you please address this issue and ensure that the list updates seamlessly without the need for manual page refreshing? Your attention to this matter would be greatly appreciated. 👍
> 
> [screen-capture (2).webm](https://github.com/FilbertKimasa/book-store/assets/40009354/0c512663-8961-4ab6-98de-65aea4529115)
>

**Review by Johnadibe — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES ♻️
### Hi @FilbertKimasa👋🏽, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

mario_ladders

## Highlights
✅ You used GitFlow correctly 👍🏽
✅ Made use of the Bookstore API as a backend✔
✅ Used Axios with createAsyncThunk to make HTTP requests to the backend✔
✅ Fetched books from the Bookstore API after the component has rendered✔
✅ Refactored add book and remove book features to persist changes in the server✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @FilbertKimasa,

#### Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

### Congratulations! 🎉 

giphy (1)

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**